### ISSUE TITLE
RFE: support loading config from `.config/` subdir

### ISSUE THREAD

**Issue by scop** [dprint/dprint#938]
There are at least a couple of initiatives related to uncluttering project top level dirs with configuration files for various tools by promoting use of the `.config/` subdir.

- https://github.com/pi0/config-dir (see also https://github.com/pi0/config-dir/discussions/6)
- https://dot-config.github.io

These are not that popular yet, but it would be great if dprint could implement this, to help unclutter on its own part and help the initiatives gain general traction.

Basically, I think just searching for the filenames dprint currently searches to find its config, but in a `.config/` subdir in addition to the current dir (including when traversing upward) would be a nice implementation.

**Comment by dsherret** [comment]
I'm not sure about doing this because there's a startup and traversal performance cost to need to search for `.config` sub directories in addition to the existing files. I think the way to bury the config file at the moment is to make it hidden like `.dprint.json`

**Comment by lonix1** [comment]
Maybe it can search all the "cheap" locations first (i.e. `./dprint.ext`), then the "expensive" `.config/dprint.ext` last.

(On another note, how much traversal is there, given it's only one fixed alternate location?)

**Comment by scop** [comment]
Dot prefix burying does not really work, as dot files are not hidden these days in a lot of contexts that matter.
